var/spack/repos/builtin/packages/superlu-dist/package.py

```diff
@@ -89,8 +89,8 @@ def append_from_variant(*args):
         append_define("CMAKE_CXX_COMPILER", spec["mpi"].mpicxx)
         append_define("CMAKE_INSTALL_LIBDIR", self.prefix.lib)
         append_define("CMAKE_INSTALL_BINDIR", self.prefix.bin)
-        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs)
-        append_define("TPL_LAPACK_LIBRARIES", spec["lapack"].libs)
+        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs.ld_flags)
+        append_define("TPL_LAPACK_LIBRARIES", spec["lapack"].libs.ld_flags)
```

Comment:
```
balay@p1 /home/balay/git-repo/github/spack (develop =)
$ git grep TPL_BLAS_LIBRARIES
var/spack/repos/builtin/packages/butterflypack/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/gptune/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/strumpack/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/superlu-dist/package.py:        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs)
```

Perhaps using `libs.joined(";")` is the right thing to do?

Reply:
But yeah - I do see `libs.ld_flags` usage - so perhaps this is ok. Will leave this for @lingnanyuan @pghysels 
```
            "TPL_PARMETIS_LIBRARIES", [spec["parmetis"].libs.ld_flags, spec["metis"].libs.ld_flags]
```

Reply:
MKL static requires some extra switches that are not libraries:
https://github.com/spack/spack/blob/a9538f2c2985558ed4863bcd3f64ac3b1db4cbde/lib/spack/spack/build_systems/oneapi.py#L203